src/env_model_repository.cpp

```diff
@@ -47,6 +47,35 @@ void EnvironmentModelRepository::updateItem(const std::string& name, std::shared
   items[name]->setPayload(plptr);
 }
 
+void EnvironmentModelRepository::removeItem(const std::string& name)
+{
+  std::shared_ptr<Item> to_be_erased = items[name];
```

Comment:
the program will crash if the `items` map does not contain `name`. 

- either wrap this with a `try catch` block and use `items.at(name)` (because `.at()` throws an exception but `[]` operator does not)
- or use `auto to_be_erased = items.find(name)`, which returns an iterator and then you can check if `to_be_erased != items.end()`. also you wouldn't be making an unnecessary copy of the item (currently you are instantiating a new shared pointer of the item you actually want to remove)